L1Trigger/RPCTriggerPrimitives/src/PrimitivePreprocess.cc

```diff
@@ -0,0 +1,160 @@
+#include "L1Trigger/RPCTriggerPrimitives/interface/PrimitivePreprocess.h"
+
+PrimitivePreprocess::PrimitivePreprocess(const edm::ParameterSet& iConfig, edm::ConsumesCollector&& iConsumes):
+  rpcToken_(iConsumes.consumes<RPCDigiCollection>(iConfig.getParameter<edm::InputTag>("Primitiverechitlabel"))),
+  processorvector_(),
+  Mapsource_(iConfig.getParameter<std::string>("Mapsource")),
+  ApplyLinkBoardCut_(iConfig.getParameter<bool>("ApplyLinkBoardCut")),
+  LinkBoardCut_(iConfig.getParameter<int>("LinkBoardCut")),
+  ClusterSizeCut_(iConfig.getParameter<int>("ClusterSizeCut")),
+  theRPCMaskedStripsObj(nullptr),
+  theRPCDeadStripsObj(nullptr),
+  maskSource_(MaskSource::EventSetup), 
+  deadSource_(MaskSource::EventSetup){
+  
+  // Get the concrete reconstruction algo from the factory
+  const std::string theAlgoName = iConfig.getParameter<std::string>("recAlgo");
+  theAlgorithm.reset(PrimitiveAlgoFactory::get()->create(theAlgoName,
+							 iConfig.getParameter<edm::ParameterSet>("recAlgoConfig")));
```

Comment:
seems like this is where the compilation fails.  please fix.

Reply:
The reconstruction algorithm is now well initialized